apps/dashboard/src/components/subscribers/locale-select.tsx

```diff
@@ -1,71 +1,111 @@
-import { RiEarthLine } from 'react-icons/ri';
-import { CountryFlags } from '../icons/country-flags';
-import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '../primitives/select';
+import { cn } from '@/utils/ui';
+import { RiArrowDownSLine, RiCheckLine, RiEarthLine } from 'react-icons/ri';
+import { type Country } from 'react-phone-number-input';
+import flags from 'react-phone-number-input/flags';
+import { Button } from '../primitives/button';
+import { Command, CommandEmpty, CommandGroup, CommandInput, CommandItem, CommandList } from '../primitives/command';
+import { Popover, PopoverContent, PopoverTrigger } from '../primitives/popover';
+import { ScrollArea } from '../primitives/scroll-area';
 import TruncatedText from '../truncated-text';
+import { useState } from 'react';
 
 export function LocaleSelect({
   value,
-  defaultOption,
+  onChange,
   disabled,
-  onValueChange,
   readOnly,
-  required,
 }: {
   value?: string;
-  defaultOption?: string;
-  size?: 'sm' | 'md';
   disabled?: boolean;
   readOnly?: boolean;
-  required?: boolean;
-  onValueChange: (val: string) => void;
+  onChange: (val: string) => void;
 }) {
+  const [open, setOpen] = useState(false);
+  const currentCountryCode = value?.split('_')?.[1] as Country;
+  const CurrentFlag = currentCountryCode ? flags[currentCountryCode] : RiEarthLine;
+
   return (
-    <Select
-      value={value}
-      onValueChange={onValueChange}
-      disabled={disabled || readOnly}
-      required={required}
-      defaultValue={defaultOption}
-    >
-      <SelectTrigger className="focus:ring-stroke-strong group p-1.5 shadow-sm focus:ring-1">
-        <SelectValue
-          placeholder={
-            <div className="flex w-full items-center gap-2">
-              <div>
-                <RiEarthLine className="size-4" />
-              </div>
-              <TruncatedText className="text-sm">Locale</TruncatedText>
-            </div>
-          }
-          asChild
+    <Popover open={open} onOpenChange={setOpen}>
+      <PopoverTrigger asChild>
+        <Button
+          type="button"
+          variant="secondary"
+          mode="outline"
+          className="flex h-8 w-full items-center justify-between gap-1 rounded-lg border-r-0 px-3 focus:z-10"
+          disabled={disabled}
         >
-          <div className="flex w-full items-center gap-2">
-            {value && (
-              <>
-                <div>
-                  <CountryFlags name={value.split('_')?.[1]} className="size-4" />
-                </div>
-                <TruncatedText className="text-sm">{value}</TruncatedText>
-              </>
-            )}
+          <div className="flex max-w-full flex-1 items-center gap-1 overflow-hidden">
+            <span className="inline-flex gap-1">
+              {CurrentFlag && <CurrentFlag title={currentCountryCode} className="size-4" />}
+              {value || <span className="text-neutral-400">locale</span>}
```

Comment:
IDK why but the locale dropdown font size is bigger that the timezone, we should use the same font across the board